### PR TITLE
Fix explanation of two-digit year handling in JS date methods

### LINKED ISSUE #1461: Are there really two different ways of handling two-digit years?
@nickshanks commented on [Wed Nov 04 2020](https://github.com/mdn/sprints/issues/3860)

## URL(s)
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#Compatibility_notes
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#Two_digit_years_map_to_1900_–_1999
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setYear#Description

## Request type

- [ ] New documentation
- [X] Correction or update

## Details
The Date class page and the setYear page both say two-digit years are parsed as an offset to the year 1900 (formerly, in the H3: "Two digit years map to 1900 – 1999")  
The Date.parse page says that `Date.parse()` and `new Date()` use values 50-99 to mean 1950-1999 and 0-49 to mean 2000-2049. Thus this apparently differs from how `Date.prototype.setYear()` handles two-digit years.

Are the two implementations of two-digit-year handling actually divergent? If so, why isn't this made explicit in the documentation for both functions? If not, I presume the documentation for setYear(), on both the setYear page and the class page, is incorrect.

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/1461

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/date/setyear/index.md [hunk 1]
```diff
@@ -12,10 +12,15 @@ browser-compat: javascript.builtins.Date.setYear
...
 
-The **`setYear()`** method sets the year for a specified date
-according to local time. Because `setYear()` does not set full years ("year
-2000 problem"), it is no longer used and has been replaced by the
-{{jsxref("Date.prototype.setFullYear()", "setFullYear()")}} method.
+The legacy **`setYear()`** method sets the year for a specified date according to local time.
+
+However, the way the legacy `setYear()` method sets year values is different from how the preferred {{jsxref("Date.prototype.setFullYear()", "setFullYear()")}} method sets year values — and in some cases, also different from how `new Date()` and {{jsxref("Date.parse()")}} set year values. Specifically, given two-digit numbers, such as `22` and `61`:
+
+* `setYear()` interprets any two-digit number as an offset to `1900`; so `date.setYear(22)` results in the year value being set to `1922`, and `date.setYear(61)` results in the year value being set to `1961`. (In contrast, while `new Date(61, 1)` also results in the year value being set to `1961`, `new Date(22, 1)` results in the year value being set to `2022`; and similarly for {{jsxref("Date.parse()")}}).
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/date/setyear/index.md:19]: The clause:

_`new Date(22, 1)` results in the year value being set to `2022`;_

 contradicts the example above (line 182) where you wrote:

`let date = new Date(22, 1)         // Wed Feb 01 1922 00:00:00 GMT+0000 (GMT)`

**Review by teoli2003 — CHANGES_REQUESTED** [review]
There is one contradiction, but else it is much better!

File: files/en-us/web/javascript/reference/global_objects/date/setyear/index.md (hunk 1 above)
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/date/setyear/index.md:19]:
```suggestion
* `setYear()` interprets any two-digit number as an offset to `1900`; so `date.setYear(22)` results in the year value being set to `1922`, and `date.setYear(61)` results in the year value being set to `1961`. (In contrast, while `new Date(61, 1)` also results in the year value being set to `1961`, `new Date("2/1/22")` results in the year value being set to `2022`; and similarly for {{jsxref("Date.parse()")}}).
```

Comment (sideshowbarker) [reply files/en-us/web/javascript/reference/global_objects/date/setyear/index.md]: > The clause:
> 
> _`new Date(22, 1)` results in the year value being set to `2022`;_
> 
> contradicts the example above (line 182) where you wrote:
> 
> `let date = new Date(22, 1) // Wed Feb 01 1922 00:00:00 GMT+0000 (GMT)`

Oops, thanks for catching that; fixed now

**Review by teoli2003 — APPROVED** [review]

**Pull request merged**